File: mobile_app/lib/models/restaurant.dart
class Restaurant {
  final int id;
  final String name;
  final String description;
  final String address;
  final double latitude;
  final double longitude;

  Restaurant({
    required this.id,
    required this.name,
    required this.description,
    required this.address,
    required this.latitude,
    required this.longitude,
  });

  factory Restaurant.fromJson(Map<String, dynamic> json) {
    return Restaurant(
      id: int.tryParse('${json['id']}') ?? 0,
      name: json['name'] as String? ?? '',
      description: json['description'] as String? ?? '',
      address: json['address'] as String? ?? '',
      latitude: double.tryParse('${json['latitude'] ?? 0}') ?? 0.0,
      longitude: double.tryParse('${json['longitude'] ?? 0}') ?? 0.0,
    );
  }
}

File: common/models/Restaurant.php
<?php
namespace common\models;

use yii\db\ActiveRecord;

class Restaurant extends ActiveRecord
{
    public static function tableName()
    {
        return 'restaurants';
    }

    public function rules()
    {
        return [
            [['owner_id', 'name', 'status'], 'required'],
            ['name', 'string', 'max' => 255],
            ['status', 'in', 'range' => ['draft', 'pending', 'active', 'suspended', 'closed']],
            [['latitude', 'longitude'], 'number'],
        ];
    }

    public function fields()
    {
        $fields = parent::fields();
        $fields['is_verified'] = function () {
            $user = User::findOne($this->owner_id);
            return $user && $user->status === 'active' && $this->status === 'active';
        };
        return $fields;
    }
}